src/index.ml

```diff
@@ -298,7 +375,7 @@ struct
                 append_key_value log.io e.key e.value)
               io;
             IO.sync log.io;
-            IO.clear io)
+            IO.clear ~generation:0L io)
```

Comment:
I believe you're correctly preserving the existing semantics, but why don't we increment the generation number here?

Reply:
We are only increasing the generation on clear when calling `clear` in index (as opposed to whenever we call `IO.clear`). 
I changed this line with the last commit, so that `log_async` uses the same generation number as index and log. But it doesn't change too much, we never read the generation from the log_async.

Reply:
OK :+1: Thanks for the explanation. This will make more sense in the code when we have better distinction between IO and header formats.